server/message_hooks_test.go

```diff
@@ -178,3 +179,256 @@ func TestReactionHasBeenAdded(t *testing.T) {
+				uclient.On("CreateOrGetChatForUsers", mock.Anything).Return(testutils.GetID(), nil).Times(1)
+				uclient.On("UpdateChatMessage", testutils.GetMockArgumentsWithType("string", 3)...).Return(nil).Times(1)
+				uclient.On("GetChatMessage", testutils.GetMockArgumentsWithType("string", 2)...).Return(&msteams.Message{}, nil).Times(1)
+			},
+		},
+		{
+			Name: "MessageHasBeenUpdated: Unable to get the link by channel ID and channel",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetUser", testutils.GetID()).Return(testutils.GetUser(model.SystemAdminRoleId, "test@test.com"), nil).Times(1)
+				api.On("GetChannel", testutils.GetChannelID()).Return(nil, testutils.GetInternalServerAppError("unable to get the channel")).Times(1)
+			},
+			SetupStore: func(store *storemocks.Store) {
+				store.On("GetLinkByChannelID", testutils.GetChannelID()).Return(nil, nil).Times(1)
+				store.On("GetTokenForMattermostUser", testutils.GetID()).Return(&oauth2.Token{}, nil).Times(1)
+			},
+			SetupClient: func(client *clientmocks.Client, uclient *clientmocks.Client) {},
+		},
+		{
+			Name: "MessageHasBeenUpdated: Unable to get the link by channel ID and channel type is Open",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetUser", testutils.GetID()).Return(testutils.GetUser(model.SystemAdminRoleId, "test@test.com"), nil).Times(1)
+				api.On("GetChannel", testutils.GetChannelID()).Return(testutils.GetChannel(model.ChannelTypeOpen), nil).Times(1)
+			},
+			SetupStore: func(store *storemocks.Store) {
+				store.On("GetLinkByChannelID", testutils.GetChannelID()).Return(nil, nil).Times(1)
+				store.On("GetTokenForMattermostUser", testutils.GetID()).Return(&oauth2.Token{}, nil).Times(1)
+			},
+			SetupClient: func(client *clientmocks.Client, uclient *clientmocks.Client) {},
+		},
+		{
+			Name: "MessageHasBeenUpdated: Unable to get the link by channel ID and unable to get channel members",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetUser", testutils.GetID()).Return(testutils.GetUser(model.SystemAdminRoleId, "test@test.com"), nil).Times(1)
+				api.On("GetChannel", testutils.GetChannelID()).Return(testutils.GetChannel(model.ChannelTypeDirect), nil).Times(1)
+				api.On("GetChannelMembers", testutils.GetChannelID(), 0, 10).Return(nil, testutils.GetInternalServerAppError("unable to get channel members")).Times(1)
+			},
+			SetupStore: func(store *storemocks.Store) {
+				store.On("GetLinkByChannelID", testutils.GetChannelID()).Return(nil, nil).Times(1)
+				store.On("GetTokenForMattermostUser", testutils.GetID()).Return(&oauth2.Token{}, nil).Times(1)
+			},
+			SetupClient: func(client *clientmocks.Client, uclient *clientmocks.Client) {},
+		},
+		{
+			Name: "MessageHasBeenUpdated: Unable to get the link by channel ID and unable to update the chat",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogError", mock.AnythingOfType("string"), "error", mock.Anything).Times(2)
```

Comment:
I think if this emits multiple errors, I would be specific about that errors, adding 2 lines but with the error properly specified there.